sktime/transformations/series/tests/test_kalman_filter.py

```diff
@@ -892,3 +907,110 @@ def test_transform_and_smooth_fp(params, measurements, y):
         Xs=means, Ps=covs, Fs=matrices["Fs"], Qs=matrices["Qs"]
     )[0]
     assert_array_almost_equal(xt_smoother_adapter, xt_smoother_filterpy)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class([KalmanFilterTransformerPK, KalmanFilterTransformerSIMD]),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+@pytest.mark.parametrize("hidden", [True, False])
+@pytest.mark.parametrize("denoising", [True, False])
+@pytest.mark.parametrize(
+    "params, measurements",
+    [
+        # test case 1 -
+        #   state_dim = 2, measurement_dim = 1, params - params_2_1_static
+        (dict(params_2_1_static, state_dim=2), create_data((ts, 1))),
+        # test case 2 -
+        #   state_dim = 3, measurement_dim = 3, params - params_3_3_static
+        (dict(params_3_3_static, state_dim=3), create_data((ts, 3))),
+    ],
+)
+def test_transform_and_smooth_simd(hidden, denoising, params, measurements):
+    """Test KalmanFilterTransformerSIMD `fit` and `transform`.
+
+    Compare the result with `pykalman`'s `filter` or `smooth`.
+    """
+    mask_measurements = np.ma.masked_invalid(np.copy(measurements))
+
+    extended_params = dict(params, hidden=hidden, denoising=denoising)
+
+    # adapter transformer
+    adapter_transformer = KalmanFilterTransformerSIMD(**extended_params)
+    xt_adapter_transformer = adapter_transformer.fit_transform(measurements)
+
+    # pykalman
+    kf_pykalman = init_kf_pykalman(measurement_dim=measurements.shape[1], **params)
+    if "estimate_matrices" in params.keys():
+        kf_pykalman = kf_pykalman.em(mask_measurements)
+
+    if denoising:
+        xt_pykalman, _ = kf_pykalman.smooth(mask_measurements)
+    else:
+        xt_pykalman, _ = kf_pykalman.filter(mask_measurements)
+
+    if not hidden:
+        meas_dim = measurements.shape[1]
+        state_dim = extended_params["state_dim"]
+        H = extended_params.get("measurement_function", np.eye(meas_dim, state_dim))
+        xt_pykalman = np.dot(xt_pykalman, H.T)
+
+    # test filter()
+    assert_array_almost_equal(xt_adapter_transformer, xt_pykalman)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(KalmanFilterTransformerSIMD),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+@pytest.mark.parametrize("smooth", [True, False])
```

Comment:
tests should be in the `tests` folder, module names starting with `test`.

Reply:
Do you mean that this file should now be moved from `series/tests/` to `series/kalman_filter/tests/`?

Reply:
I might have imagined this being at the end of a non-test file. Either place is good, but `kalman_filter/tests` is slightly better.